pkg/microservice/aslan/core/service/service/environment.go

```diff
@@ -46,6 +56,217 @@ func GetDeployableEnvs(svcName, projectName string) ([]string, error) {
+	kubeClient, err := kubeclient.GetKubeClient(config.HubServerAddress(), params.ClusterID)
+	if err != nil {
+		log.Errorf("cluster is not connected [%s]", params.ClusterID)
+		return err
+	}
+	for _, service := range params.Services {
+		var yamls []string
+		for workloadType, workloads := range service.WorkloadsMap {
+			switch workloadType {
+			case "configmap":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetConfigMapYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return e.ErrGetService.AddDesc(fmt.Sprintf("get deploy/configmap failed err:%s", err))
+					}
+
+					yamls = append(yamls, string(bs))
+				}
+			case "deployment":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetDeploymentYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return e.ErrGetService.AddDesc(fmt.Sprintf("get deploy/deployment failed err:%s", err))
+					}
+
+					yamls = append(yamls, string(bs))
+				}
+			case "service":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetServiceYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return e.ErrGetService.AddDesc(fmt.Sprintf("get deploy/service failed err:%s", err))
+					}
+
+					yamls = append(yamls, string(bs))
+				}
+			case "secret":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetSecretYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return e.ErrGetService.AddDesc(fmt.Sprintf("get deploy/secret failed err:%s", err))
+					}
+
+					yamls = append(yamls, string(bs))
+				}
+			case "ingress":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetIngressYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return e.ErrGetService.AddDesc(fmt.Sprintf("get deploy/ingress failed err:%s", err))
+					}
+
+					yamls = append(yamls, string(bs))
+				}
+			case "statefulset":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetStatefulSetYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return fmt.Errorf("get deploy/statefulset failed err:%s", err)
+					}
+					yamls = append(yamls, string(bs))
+				}
+			case "pvc":
+				for _, workload := range workloads {
+					bs, _, err := getter.GetPVCYamlFormat(params.Namespace, workload, kubeClient)
+					if len(bs) == 0 || err != nil {
+						log.Errorf("not found yaml %v", err)
+						return fmt.Errorf("get deploy/pvc failed err:%s", err)
+					}
+					yamls = append(yamls, string(bs))
```

Comment:
is there anyway to write a function like getter.GetWorkloadYamlFormat and pass workload type as a parameter to reduce duplicate code？

Reply:
no